apps/accelerate/chatllama/chatllama/rlhf/actor.py

```diff
@@ -54,93 +56,97 @@ def __init__(self, config: ConfigActor) -> None:
-            if self.tokenizer.eos_token is None:
-                self.tokenizer.eos_token = "</s>"
-                self.tokenizer.eos_token_id = 0
-            self.tokenizer.pad_token = self.tokenizer.eos_token
-            self.tokenizer.pad_token_id = self.tokenizer.eos_token_id
+            self.tokenizer = self.load_tokenizer(config)
             self.model = AutoModelForCausalLM.from_pretrained(
                 config.model,
             )
             self.model.to(config.device)
+        else:
+            raise ValueError(f"Model {config.model} not supported")
 
         # load the model from model_folder
         self.load()
 
     @beartype
     def load(self) -> None:
-        """Load the model from the path
-
-        Args:
-            path (str): path to the model
-        """
+        """Load the model from the path"""
+        # check if there is a model to load
         path = ModelLoader.check_model_path(
             config=self.config,
             is_checkpoint=False,
             current_epoch=None,
         )
+
+        # if there is a model to load
         if path is not None:
+
+            # load the model
             print("Loading ...")
             model_dict = torch.load(path)
-            self.model.load_state_dict(model_dict["model"])
-            self.head.load_state_dict(model_dict["head"])
+            self.model.load_state_dict(model_dict["state_dict"])
 
     @beartype
     def save(self) -> None:
-        """Save the model to the path
-
-        Args:
-            path (Optional[str], optional): Path to store the model.
-                Defaults to None.
-        """
+        """Save the model to the path"""
+        # get the path to save the model
         model_folder, model_name, path = ModelLoader.get_model_path(
             config=self.config,
             is_checkpoint=False,
             current_epoch=None,
         )
+
+        # save the model
         print(f"Saving model to {path} ...")
         torch.save(
-            {"model": self.model.state_dict(), "head": self.head.state_dict()},
+            {"state_dict": self.model.state_dict()},
             path,
         )
 
-    def parameters(self, **kwargs):
-        """Return the parameters of the model
+    @staticmethod
+    def load_tokenizer(config: ConfigActor):
+        """Load the tokenizer from the model name"""
+        if config.model in hf_models:
+            # load the tokenizer from HF
+            tokenizer = AutoTokenizer.from_pretrained(
+                config.model,
+                padding_side="left",
+                padding=True,
+                truncation=True,
+                model_max_length=config.max_sequence_length,
+            )
 
-        Args:
-            **kwargs:
-        """
+            # add eos token if not present
+            if tokenizer.eos_token is None:
+                tokenizer.eos_token = "</s>"
+                tokenizer.eos_token_id = 0
```

Comment:
It would be great... How can i known what id is significant? 
It depends on the model and there was only one model that I think Galactica that has these problem of eos not defined.